### PR TITLE
Update nixpkgs to track nixos-21.11 branch and update other inputs

### LINKED ISSUE #5: Build fails under latest unstable-small branch
Hey,

It seems that one of the dependencies of this project uses `stdenv.lib` (I've reported it to `nixpkgs-mozilla` already, hopefully a fix comes soon), and as such prevents building on the current `unstable-small` branch. The current workaround is to force this to follow stable, since the alias still exists.

### PR DESCRIPTION
**Pull request by astratagem**
Fixes #5

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   description = "agenix-cli";
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-20.09";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/release-21.11";
```
Comment (cole-h) [inline flake.nix]: The release branch is what commits get merged into -- so it's basically the "master" branch of the 21.11 release, which is not reallly what we want.

```suggestion
  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-21.11";
```

**Review by cole-h — CHANGES_REQUESTED** [review]
Thanks for the PR! Just one minor request:

File: flake.nix (hunk 1 above)
Comment (astratagem) [reply flake.nix]: Ah, thanks! I didn't realize that.

**Review by cole-h — APPROVED** [review]
Diff LGTM, thanks!

I can't test this right now, so I'll merge it tomorrow (provided all goes well).

Also cc @enderger -- does this PR fix your issue from #5?

**Comment by cole-h** [comment]
Everything seems to still be working. Thanks!

**Pull request merged**